MyHashMap.java

```diff
@@ -1 +1,157 @@
+                    currentNode.setValue(value);
+                } else {
+                    currentNode = currentNode.next;
+                }
+            }
+            nodes[index] = new Node(null, currentNode.getKey(), currentNode.getValue());
+            countOfElem++;
+        } else {
+            if (Objects.isNull(key) && Objects.nonNull(value) || currentNode.getKey().equals(key)) {
+                nodes[index] = new Node(null, currentNode.getKey(), currentNode.getValue());
+                countOfElem++;
+            } else {
+                throw new NoSuchElementException("The key " + key + " is not found.");
+            }
+        }
+    }
+
+    @Override
+    public V get(K key) {
+        for (int i = 0; i < nodes.length; i++) {
+            if (Objects.nonNull(nodes[i])) {
+                if (Objects.nonNull(key) && Objects.nonNull(nodes[i].getKey()) && nodes[i].getKey().equals(key) ||
+                        Objects.isNull(key) && Objects.isNull(nodes[i].getKey())) {
+                    return nodes[i].getValue();
+                }
+            }
+        }
+        throw new NoSuchElementException("The value with key " + key + " is not found.");
+    }
+
+    @Override
+    public void increase() {
+        Node<K, V>[] newNode = new Node[newCapacity()];
+        for (int i = 0; i < nodes.length; i++) {
+            if (Objects.nonNull(nodes[i])) {
+                K oldKey = nodes[i].getKey();
+                int index = indexFor(oldKey, newNode.length);
+                if (Objects.isNull(newNode[index])) {
+                    newNode[index] = nodes[i];
+                } else {
+                    K newKey = newNode[index].getKey();
+                    while (Objects.nonNull(nodes[i].next)) {
+                        if (Objects.isNull(newNode[index])) {
+                            newNode[index] = nodes[i];
+                            nodes[i] = nodes[i].next;
+                        } else {
+                            boolean hasNext = newNode[index].hasNext();
+                            while (hasNext) {
+                                K nextKey = newNode[index].next.getKey();
+                                if (newKey.equals(nextKey)) {
+                                    newNode[index].next.setValue(newNode[index].value);
+                                    hasNext = false;
+                                } else {
+                                    newNode[index] = newNode[index].next;
+                                }
+                            }
+                            newNode[index] = new Node(null, nodes[i].getKey(), nodes[i].getValue());
+                        }
+                        countOfElem++;
+                    }
+                }
+            }
+        }
+        nodes = Arrays.copyOf(newNode, newCapacity());
+    }
+
+    private int newCapacity() {
+        return nodes.length * 2;
+    }
+
+    @Override
+    public void clear() {
```

Comment:
better remove this method if we don't have implementation

Reply:
Yes, i forgot him